tasks/utilities/task_validate.wdl

```diff
@@ -2,54 +2,284 @@ version 1.0
 
 task export_two_tsvs {
   input {
-    String terra_project
-    String terra_workspace
+    String terra_project1
+    String? terra_project2
+    String terra_workspace1
+    String? terra_workspace2
     String datatable1
     String datatable2
     Int disk_size = 10
   }
   command <<<
-    python3 /scripts/export_large_tsv/export_large_tsv.py --project ~{terra_project} --workspace ~{terra_workspace} --entity_type ~{datatable1} --tsv_filename ~{datatable1}
+    python3 /scripts/export_large_tsv/export_large_tsv.py --project ~{terra_project1} --workspace ~{terra_workspace1} --entity_type ~{datatable1} --tsv_filename ~{datatable1}
 
-    python3 /scripts/export_large_tsv/export_large_tsv.py --project ~{terra_project} --workspace ~{terra_workspace} --entity_type ~{datatable2} --tsv_filename ~{datatable2}
+    # check if second project is provided; if not, use first
+    if [[ -z "~{terra_project2}" ]]; then
+      PROJECT2="~{terra_project1}"
+    else
+      PROJECT2="~{terra_project2}"
+    fi
+
+    # check if second workspace is provided; if not, use first
+    if [[ -z "~{terra_workspace2}" ]]; then
+      WORKSPACE2="~{terra_workspace1}"
+    else
+      WORKSPACE2="~{terra_workspace2}"
+    fi
+
+    python3 /scripts/export_large_tsv/export_large_tsv.py --project ${PROJECT2} --workspace ${WORKSPACE2} --entity_type ~{datatable2} --tsv_filename ~{datatable2}
+
+    if [[ $(wc -l ~{datatable1} | cut -f1 -d' ') -eq $(wc -l ~{datatable2} | cut -f1 -d' ') ]]; then
+      echo true | tee CONTINUE
+    else 
+      echo false | tee CONTINUE
+    fi
   >>>
   runtime {
     docker: "quay.io/theiagen/terra-tools:2023-03-16"
     memory: "1 GB"
     cpu: 1
     disks:  "local-disk " + disk_size + " HDD"
-    disk: disk_size + " GB" # TES
+    disk: disk_size + " GB"
     dx_instance_type: "mem1_ssd1_v2_x2"
-    maxRetries: 3
   }
   output {
     File datatable1_tsv = "~{datatable1}"
     File datatable2_tsv = "~{datatable2}"
+    Boolean same_table_length = read_boolean("CONTINUE")
   }
 }
 
 task compare_two_tsvs {
   input {
+    String datatable1
     File datatable1_tsv
+    String datatable2
     File datatable2_tsv
-    String? out_dir
-    String? out_prefix
+    File? validation_criteria_tsv
+    String columns_to_compare
+    String output_prefix
+
     Int disk_size = 10
   }
   command <<<
-    compare-data-tables.py ~{datatable1_tsv} ~{datatable2_tsv} --outdir ~{out_dir} --prefix ~{out_prefix}
+    # too lazy to create a new docker image, this is not good practice
```

Comment:
Hah! I literally just wrote this out before seeing this comment: 

> python block is getting a bit complex; we should consider storing this into a separate repo and building a docker for these types of utilities

Reply:
My one problem with making it into a script included in a docker is that whenever we want to make updates to the code, we'll have to make a new docker image as well. That's not a deal-breaker, but it's an extra step and an extra level of obscurity when people have to figure out where to find the code that performs the comparisons. 

A large part of this python block is just comments explaining what's going on as well